File: src/data/lessons/nurseryLessons.js
import { Lesson } from '../../models/Lesson.js';

/**
 * Nursery Lessons
 */
export function getNurseryLessons(startLessonId, startQuizId) {
  let lessonId = startLessonId;
  let quizId = startQuizId;

  return [
    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 1,
      title: "Clicking Game",
      emoji: '🎯',
      assessmentType: 'clicking-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 2,
      title: "Keyboard Game",
      emoji: '⌨️',
      assessmentType: 'keyboard-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 3,
      title: "WASD Game",
      emoji: '⌨️',
      assessmentType: 'wasd-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 4,
      title: "A-Z Game",
      emoji: '🔤',
      assessmentType: 'a-z-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 5,
      title: "Numbers Game",
      emoji: '🔢',
      assessmentType: 'numbers-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 6,
      title: "Symbols Game",
      emoji: '🔣',
      assessmentType: 'symbols-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'technology',
      lessonNumber: 7,
      title: "Flappy Bird Game",
      emoji: '🐦',
      assessmentType: 'flappy-bird-game',
      categoryId: null,
    }),

    // Phonics Lessons for 2-3 year olds - Lessons 1-8

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'history',
      lessonNumber: 1,
      title: "Dinosaurs - The First Animals",
      emoji: '🦕',
      content: `# Dinosaurs - The First Animals 🦕



Long, long ago, before people lived, there were dinosaurs!



## What are Dinosaurs?



Dinosaurs were huge animals that lived a very, very long time ago!



- They were bigger than elephants! 🦕

- Some were very tall

- Some were very long

- They lived millions of years ago



## Big Dinosaurs



**T-Rex** 🦖

- Very big and strong

- Had sharp teeth

- Was a meat eater

- Very scary!



**Brachiosaurus** 🦕

- Had a very long neck

- Was very tall

- Ate plants

- Was gentle



## Small Dinosaurs



- Some dinosaurs were small

- Some were as big as chickens

- They all lived together

- Long, long ago!



## Fun Activities



- Draw pictures of dinosaurs

- Make dinosaur sounds

- Learn dinosaur names

- Pretend to be a dinosaur!



## Remember



- Dinosaurs lived long ago

- They were very big animals

- They don't live anymore

- We learn about them from fossils!



## Practice Questions



<!-- QUESTION_START -->
When did dinosaurs live?
<!-- OPTIONS -->
Today|A very, very long time ago, millions of years ago|Last year|Next year
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Dinosaurs lived a very, very long time ago - millions of years ago! They were bigger than elephants!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
What did T-Rex eat?
<!-- OPTIONS -->
Plants|Meat|Nothing|Fruits
<!-- CORRECT -->
1
<!-- EXPLANATION -->
T-Rex was a meat eater! It was very big and strong, had sharp teeth, and was very scary!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
What did Brachiosaurus eat?
<!-- OPTIONS -->
Meat|Plants|Nothing|Fruits
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Brachiosaurus ate plants! It had a very long neck, was very tall, and was gentle!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
How do we learn about dinosaurs?
<!-- OPTIONS -->
We see them|From fossils|We don't|We guess
<!-- CORRECT -->
1
<!-- EXPLANATION -->
We learn about dinosaurs from fossils! Even though they don't live anymore, we can still learn about them!
<!-- QUESTION_END -->`,
      quizId: 31,
      assessmentType: 'history-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'history',
      lessonNumber: 2,
      title: "Adam and Eve - The First People",
      emoji: '👫',
      content: `# Adam and Eve - The First People 👫



Let's learn about the first people in the world!



## The Garden of Eden



- Adam and Eve lived in a beautiful garden 🌳

- The garden was called Eden

- It was a perfect place

- Everything was good



## Adam and Eve



- Adam was the first man 👨

- Eve was the first woman 👩

- They were the first people

- They lived in the garden together



## The First Family



- Adam and Eve were together

- They took care of the garden

- They were happy

- They were the first family!



## Fun Activities



- Draw the garden of Eden

- Talk about the first people

- Learn about the story

- Draw Adam and Eve



## Remember



- Adam and Eve were the first people

- They lived in a beautiful garden

- They were the first family

- This is a special story!



## Practice Questions



<!-- QUESTION_START -->
Who were the first people?
<!-- OPTIONS -->
No one|Adam and Eve|Moses|Noah
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Adam and Eve were the first people! Adam was the first man, and Eve was the first woman!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
Where did Adam and Eve live?
<!-- OPTIONS -->
In a house|In a beautiful garden called Eden|In a cave|In a city
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Adam and Eve lived in a beautiful garden called Eden! It was a perfect place where everything was good!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
What did Adam and Eve do in the garden?
<!-- OPTIONS -->
Nothing|Took care of the garden and were happy together|Only played|Only slept
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Adam and Eve were together, took care of the garden, were happy, and they were the first family!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
Why is this story special?
<!-- OPTIONS -->
It's not|Adam and Eve were the first people and the first family|It's scary|It's boring
<!-- CORRECT -->
1
<!-- EXPLANATION -->
This is a special story because Adam and Eve were the first people, they lived in a beautiful garden, and they were the first family!
<!-- QUESTION_END -->`,
      quizId: 37,
      assessmentType: 'history-game',
      categoryId: null,
    }),

    new Lesson({
      id: lessonId++,
      yearId: 'nursery',
      subjectId: 'history',
      lessonNumber: 4,
      title: "Days of the Week",
      emoji: '📅',
      assessmentType: null, // Uses HTML game instead
      content: `# Days of the Week 📅



There are seven days in a week!



## The Seven Days



1. Monday - Start of the week! 🌟

2. Tuesday - Keep going! 💪

3. Wednesday - Middle of the week! 🎯

4. Thursday - Almost there! ⏰

5. Friday - Fun day! 🎉

6. Saturday - Weekend! 🎊

7. Sunday - Rest day! 😴



## What We Do Each Day



- Monday: School starts! 📚

- Tuesday: Learning new things! 🎓

- Wednesday: Mid-week fun! 🎨

- Thursday: More learning! ✏️

- Friday: End of school week! 🎈

- Saturday: Play time! 🧸

- Sunday: Family time! 👨‍👩‍👧‍👦



## Fun Activities



- Sing the days of the week song

- Point to today on a calendar

- Draw what you do each day

- Count the days until the weekend!



## Remember



- There are 7 days in a week

- Each day has a name

- Days help us know what to do!



## Practice Questions



<!-- QUESTION_START -->
How many days are in a week?
<!-- OPTIONS -->
5|7|10|12
<!-- CORRECT -->
1
<!-- EXPLANATION -->
There are 7 days in a week! Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, and Sunday!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
What day is the start of the week?
<!-- OPTIONS -->
Sunday|Monday|Friday|Saturday
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Monday is the start of the week! Then comes Tuesday, Wednesday, Thursday, Friday, Saturday, and Sunday!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
What days are the weekend?
<!-- OPTIONS -->
Monday and Tuesday|Saturday and Sunday|Wednesday and Thursday|Friday and Saturday
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Saturday and Sunday are the weekend! Saturday is play time, and Sunday is rest day and family time!
<!-- QUESTION_END -->



<!-- QUESTION_START -->
Why are days of the week important?
<!-- OPTIONS -->
They're not|Days help us know what to do and each day has a name|They're boring|They're scary
<!-- CORRECT -->
1
<!-- EXPLANATION -->
Days of the week are important because they help us know what to do! Each day has a name, and we can plan our week!
<!-- QUESTION_END -->`,
      quizId: 33,
      categoryId: null,
    }),

    

File: lib/data/lessons/nursery_lessons.dart
import '../../models/lesson.dart';

List<Lesson> getNurseryLessons(int startLessonId, int startQuizId) {
  int lessonId = startLessonId;
  int quizId = startQuizId;

  return [
    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'maths',

      lessonNumber: 1,

      title: 'Counting to 10',

      emoji: '🔢',

      content: '''

# Counting to 10



Let's learn to count from 1 to 10!



## Numbers 1-10



1️⃣ One

2️⃣ Two

3️⃣ Three

4️⃣ Four

5️⃣ Five

6️⃣ Six

7️⃣ Seven

8️⃣ Eight

9️⃣ Nine

🔟 Ten



## Practice Counting



Count along with me:

- 1, 2, 3, 4, 5, 6, 7, 8, 9, 10!



## Fun Activities



- Count your fingers! How many do you have?

- Count your toes! How many are there?

- Count objects around you: toys, books, crayons!



## Remember



- Numbers help us count things

- We start counting from 1

- 10 is the biggest number we're learning today

      ''',

      quizId: quizId++,

      assessmentType: 'quiz',

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'maths',

      lessonNumber: 2,

      title: 'Counting to 20',

      emoji: '🔢',

      content: '''

# Counting to 20



Now let's learn to count even higher - from 1 to 20!



## Numbers 1-20



1️⃣ One

2️⃣ Two

3️⃣ Three

4️⃣ Four

5️⃣ Five

6️⃣ Six

7️⃣ Seven

8️⃣ Eight

9️⃣ Nine

🔟 Ten

1️⃣1️⃣ Eleven

1️⃣2️⃣ Twelve

1️⃣3️⃣ Thirteen

1️⃣4️⃣ Fourteen

1️⃣5️⃣ Fifteen

1️⃣6️⃣ Sixteen

1️⃣7️⃣ Seventeen

1️⃣8️⃣ Eighteen

1️⃣9️⃣ Nineteen

2️⃣0️⃣ Twenty



## Practice Counting



Count along with me:

- 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20!



## Fun Activities



- Count all your fingers and toes together! (That's 20!)

- Count steps as you walk

- Count blocks as you build a tower

- Count animals in a picture book



## Remember



- After 10, we have 11, 12, 13, and so on

- 20 is a big number!

- Practice counting every day to get better

      ''',

      quizId: quizId++,

      assessmentType: 'quiz',

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'technology',

      lessonNumber: 1,

      title: 'Clicking Game',

      emoji: '🎯',

      content: '''

# Clicking Game 🎯



Welcome to the Accuracy Clicking Game!



## How to Play



- Click on the red circles as they appear on the screen

- The circles start large and get smaller and faster as time goes on

- You have 30 seconds to score as many points as possible

- Each circle you click gives you 10 points



## Scoring System



- **Bronze**: 0-99 points

- **Silver**: 100-199 points

- **Gold**: 200-299 points

- **Platinum**: 300+ points



## Ready to Play?



Click the button below to start the game!

      ''',

      quizId: null,

      assessmentType: null,

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'technology',

      lessonNumber: 2,

      title: 'Keyboard Game',

      emoji: '⌨️',

      content: '''

# Keyboard Game ⌨️



Welcome to the Keyboard Game!



## How to Play



- Watch for arrows that appear on the screen ⬆️⬇️⬅️➡️

- Press the matching key on your keyboard

- Use **WASD** keys or **Arrow Keys**

- You have 45 seconds to score as many points as possible!

- Each correct key press gives you 10 points



## Controls



- **⬆️ Up Arrow** = Press **↑** or **W**

- **⬇️ Down Arrow** = Press **↓** or **S**

- **⬅️ Left Arrow** = Press **←** or **A**

- **➡️ Right Arrow** = Press **→** or **D**



## Scoring System



- **Bronze**: 0-99 points

- **Silver**: 100-149 points

- **Gold**: 150-199 points

- **Platinum**: 200+ points



## Ready to Play?



Click the button below to start the game!

      ''',

      quizId: null,

      assessmentType: null,

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'english',

      lessonNumber: 1,

      title: 'Learning the Alphabet',

      emoji: '🔤',

      content: '''

# Learning the Alphabet 🔤



Let's learn our ABCs!



## The Alphabet Song



A, B, C, D, E, F, G

H, I, J, K, L, M, N, O, P

Q, R, S, T, U, V

W, X, Y, and Z



## Letters A-E



A is for Apple 🍎

B is for Ball ⚽

C is for Cat 🐱

D is for Dog 🐶

E is for Elephant 🐘



## Fun Activities



- Sing the alphabet song together!

- Point to letters in books

- Find letters around the house

- Trace letters with your finger



## Remember



- There are 26 letters in the alphabet

- Each letter has a name and a sound

- We use letters to make words!

      ''',

      quizId: quizId++,

      assessmentType: 'quiz',

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'english',

      lessonNumber: 2,

      title: 'Learning Letter Sounds',

      emoji: '🔊',

      content: '''

# Learning Letter Sounds 🔊



Letters make sounds! Let's learn some!



## Letter Sounds



A says "ah" like in Apple 🍎

B says "buh" like in Ball ⚽

C says "cuh" like in Cat 🐱

D says "duh" like in Dog 🐶

E says "eh" like in Elephant 🐘



## More Sounds



F says "fuh" like in Fish 🐟

G says "guh" like in Goat 🐐

H says "huh" like in Hat 🎩

I says "ih" like in Igloo 🧊

J says "juh" like in Jam 🍓



## Practice



- Make the sound for each letter

- Find things that start with each sound

- Play "I spy" with letter sounds!



## Remember



- Every letter has a sound

- Sounds help us read words

- Practice makes perfect!

      ''',

      quizId: quizId++,

      assessmentType: 'quiz',

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'english',

      lessonNumber: 3,

      title: 'Simple Words',

      emoji: '📖',

      content: '''

# Simple Words 📖



Let's learn some simple words!



## Three-Letter Words



Cat 🐱 - C-A-T

Dog 🐶 - D-O-G

Hat 🎩 - H-A-T

Sun ☀️ - S-U-N

Car 🚗 - C-A-R



## More Words



Ball ⚽ - B-A-L-L

Book 📚 - B-O-O-K

Cup ☕ - C-U-P

Pen ✏️ - P-E-N

Toy 🧸 - T-O-Y



## Fun Activities



- Read simple words together

- Point to words in picture books

- Make words with letter blocks

- Draw pictures of words



## Remember



- Words are made of letters

- Letters make sounds

- Sounds make words!

- Reading is fun!

      ''',

      quizId: quizId++,

      assessmentType: 'quiz',

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'history',

      lessonNumber: 1,

      title: 'Toys from Long Ago',

      emoji: '🧸',

      content: '''

# Toys from Long Ago 🧸



Let's learn about toys from the past!



## What is History?



History is learning about things that happened long ago!

- Long ago means a very long time before now

- People did things differently then

- We can learn from the past!



## Old Toys



Toys from long ago were different:

- Wooden dolls 🪵

- Simple spinning tops 🎡

- Hoops and sticks 🎪

- Handmade toys 🧵

- No batteries needed!



## New Toys



Toys today are different:

- Electronic toys 🎮

- Remote control cars 🚗

- Video games 💻

- Many use batteries 🔋

- Made in factories 🏭



## Comparing Old and New



- Old toys were simpler

- New toys have more features

- Both are fun to play with!

- People enjoyed toys long ago, just like we do today!



## Fun Activities



- Draw an old toy and a new toy

- Talk about your favourite toys

- Ask grown-ups about toys they had when they were little

- Make a simple toy like a paper plane!



## Remember



- Toys have changed over time

- Old toys were simpler but still fun

- History helps us understand how things change!

      ''',

      quizId: null,

      assessmentType: null,

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'history',

      lessonNumber: 2,

      title: 'Homes from Long Ago',

      emoji: '🏠',

      content: '''

# Homes from Long Ago 🏠



Let's learn about how people lived long ago!



## Homes Long Ago



Houses from long ago were different:

- Made of wood and stone 🪵

- No electricity - used candles 🕯️

- No running water - used wells 💧

- Fireplaces for cooking and warmth 🔥

- Smaller and simpler



## Homes Today



Houses today have:

- Electricity for lights 💡

- Running water from taps 🚿

- Central heating 🌡️

- Modern kitchens 🍳

- Many rooms and comforts



## How Life Was Different



Long ago:

- People cooked on fires

- No television or computers

- Traveled by walking or horses 🐴

- Made things by hand ✋

- Life was harder but simpler



## Fun Activities



- Draw an old house and a new house

- Talk about what your home has

- Imagine living without electricity

- Draw what you think a home from long ago looked like



## Remember



- Homes have changed a lot over time

- People lived differently long ago

- We have many comforts today that people didn't have then!

      ''',

      quizId: null,

      assessmentType: null,

    ),

    Lesson(

      id: lessonId++,

      yearId: 'nursery',

      subjectId: 'history',

      lessonNumber: 3,

      title: 'Kings and Queens from Long Ago',

      emoji: '👑',

      content: '''

# Kings and Queens from Long Ago 👑



Let's learn about rulers from history!



## What is a King or Queen?



- A king is a man who ruled a country 👑

- A queen is a woman who ruled a country 👸

- They lived in big castles 🏰

- They had special crowns to wear

- People listened to them



## Famous Kings and Queens



**King Henry VIII** 👑

- Lived in England long ago

- Had six wives

- Built many castles

- Was very powerful



**Queen Elizabeth I** 👸

- Was a famous queen

- Ruled England for a long time

- Was very clever and strong

- People loved her



## Life in a Castle



Kings and queens lived in castles:

- Big stone buildings 🏰

- High walls for protection

- Many rooms inside

- Servants to help them

- Knights to protect them ⚔️



## Fun Activities



- Draw a king or queen with a crown

- Draw a castle

- Pretend to be a king or queen

- Talk about what you think castle life was like



## Remember



- Kings and queens ruled countries long ago

- They lived in castles

- They were important people in history!

      ''',

      quizId: null,

      assessmentType: null,

    ),

  ];
}
